Applications/Tizen_native/CustomShortcut/src/data.c

```diff
@@ -43,6 +43,200 @@ int parse_route(const char *source, char **route, char **data) {
   return APP_ERROR_NONE;
 }
 
+void data_get_resource_path(const char *file, char *full_path, bool shared) {
+  char *root_path;
+  if (shared) {
+    root_path = app_get_shared_resource_path();
+  } else {
+    root_path = app_get_resource_path();
+  }
+
+  if (root_path == NULL) {
+    LOG_E("failed to get resource path");
+    return;
+  }
+
+  if (full_path == NULL) {
+    LOG_E("full_path is null");
+  }
+
+  snprintf(full_path, PATH_MAX, "%s%s", root_path, file);
+  LOG_D("resource path: %s", full_path);
+  free(root_path);
+}
+
+void data_get_data_path(const char *file, char *full_path) {
+  char *root_path;
+
+  root_path = app_get_data_path();
+
+  if (root_path == NULL) {
+    LOG_E("failed to get data path");
+    return;
+  }
+
+  if (full_path == NULL) {
+    LOG_E("full_path is null");
+    return;
+  }
+
+  snprintf(full_path, PATH_MAX, "%s%s", root_path, file);
+  LOG_D("data path: %s", full_path);
+  free(root_path);
+}
+
+static void _on_data_receive(ml_tensors_data_h data,
+                             const ml_tensors_info_h info, void *user_data) {
+
+  appdata_s *ad = (appdata_s *)user_data;
+
+  void *raw_data;
+  size_t data_size;
+  FILE *file;
+
+  int status = ml_tensors_data_get_tensor_data(data, 0, &raw_data, &data_size);
+  LOG_D("size of data: %ud", data_size);
+
+  if (status != ML_ERROR_NONE) {
+    LOG_E("get tensor data failed %d", status);
+    return;
+  }
+
+  status = ml_tensors_info_get_tensor_size(info, -1, &data_size);
```

Comment:
Is the data_size received from this different from `ml_tensors_data_get_tensor_data`?

Reply:
I just needed to get `raw_data` from `ml_tensors_data_get_tensor_data`  but function needed me to pass not null pointer for the data_size . So I just reused the pointer

Reply:
Just add a small comment over the `ml_tensors_data_get_tensor_data` that the value in `data_size` is ignored to avoid confusion.